frontend/src/routes/dashboard/[groupId]/+page.svelte

```diff
@@ -149,20 +155,62 @@
 
 		try {
 			for (const availabilityData of allAvailabilityData) {
-				createAvailability(currentGroupId, availabilityData);
+				createAvailability(groupId, availabilityData);
 			}
 
 			const times = allAvailabilityData
 				.map((data) => `${data.date} from ${data.start} to ${data.end}`)
 				.join(', ');
 			successMsg.set('All availabilities saved successfully ' + times);
+			console.log('GroupID', groupId);
 			console.log('Saved times:', JSON.stringify(availableTimes));
 		} catch (error) {
 			successMsg.set('Failed to save availability');
 			console.error('Failed to save availability with error', error);
 			availableTimes = {};
 		}
 	}
+
+	async function removeAvailability(selectedDay, selectedHour) {
+		const formattedHour = `${selectedHour < 10 ? `0${selectedHour}` : selectedHour}:00`;
+		const currentData = await getGroup(groupId);
+		const matchingAvailability = currentData.availabilities.find(
+			(avail) => avail.date === selectedDay && avail.start === formattedHour
+		);
+
```

Comment:
Might be easier to propagate the availability ID's directly to the clickable components, so you immediately know which one to delete and don't have to search by date. Not a blocker, though.